### DISCUSSION TITLE
Tag push not triggering action

### DISCUSSION THREAD

**Discussion by scaccogatto** [community/community#26840] [Actions]
<p>Talking about: scaccogatto/contentful-module<br>
I have this action (<a href="https://github.com/scaccogatto/contentful-module/blob/614dc86cfc5ff21c97bcad14582a22f8b12bdbba/.github/workflows/master.yml" rel="noopener nofollow ugc">https://github.com/scaccogatto/contentful-module/blob/614dc86cfc5ff21c97bcad14582a22f8b12bdbba/.github/workflows/master.yml</a>)<br>
I have these tags (<a href="https://github.com/scaccogatto/contentful-module/tags" rel="noopener nofollow ugc">https://github.com/scaccogatto/contentful-module/tags</a>)</p>
<p>But I see no triggers. What am I missing?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Adding <code>branches</code> might help:</p>
<pre><code class="lang-auto">on:
  push:
    branches:
      - '**'
    tags:
      - 'v*'
...
</code></pre>
<aside class="quote" data-post="5" data-topic="128354">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/action-does-not-run-on-master-with-tags-ignore-v/128354/5">Action does not run on master with tags-ignore v*</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="noopener nofollow ugc">The documentation</a> says: 

If you define only tags or only branches , the workflow won’t run for events affecting the undefined Git ref. 

Your workflow has an explicit rule for tags, but not for branches, so I assume that’s the reason. Try adding 
branches:
  - '**'

to the push event to build on all branch pushes.
  </blockquote>
</aside>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/scaccogatto">@scaccogatto</a>,</p>
<p>You can try like as below to see if it can work:</p>
<pre><code class="lang-yaml">on:
  push:
    tags:
      - 'v*'
</code></pre>
<p>OR</p>
<pre><code class="lang-yaml">on:
  push:
    tags:
      - 'v**'
</code></pre>
<p>In addition, how did you push these tags to the repository? Manually executed the related git commands on your local machine to push tags, or executed the related git commands in a workflow to push tags?</p>
<p>If you used a workflow, what token did you use to authenticate? A personal access token (PAT) or the <strong>GITHUB_TOKEN</strong>?</p>
<p>If you used the <strong>GITHUB_TOKEN</strong> to push tags, it is the correct behavior that the workflow was not triggered. Because (see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow" rel="noopener nofollow ugc">here</a>):</p>
<blockquote>
<p>When you use the repository’s <strong>GITHUB_TOKEN</strong> to perform tasks on behalf of the GitHub Actions app, events triggered by the <strong>GITHUB_TOKEN</strong> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>

**Comment by scaccogatto** [comment]
<p>I am pushing with git push --follow-tags.<br>
<a class="mention" href="/simran-b">@Simran-B</a> solved it! Thanks!</p>
